cockpit/IPC/install/create_link.js

```diff
@@ -1,26 +1,42 @@
-// Imports filesystem tools
 const fs = require('fs');
 const os = require('os');
+const path = require('path');
 
 // Get the target symlink dir. Assumes that is /$HOME/.local/share/cockpit
-const targetDir = `${os.homedir()}/.local/share/cockpit`;
+const targetDir = path.join(os.homedir(), '.local', 'share', 'cockpit');
+
+// Ensure the target directory exists
+if (!fs.existsSync(targetDir)) {
+    console.log(`Target directory ${targetDir} does not exist. Creating...`);
+    fs.mkdirSync(targetDir, { recursive: true });
+}
+
 // Change directory, as symlink must be run in the dir in which the symlink should reside
 process.chdir(targetDir);
 
 // Get the root module dir. Assumes we are in /install
-const moduleRootDir = __dirname.substring(0, __dirname.lastIndexOf('/'));
+const moduleRootDir = path.dirname(__dirname);
 // Gets the folder name for a symlink name
-const moduleFolderName = moduleRootDir.substring(moduleRootDir.lastIndexOf('/') + 1, moduleRootDir.length);
+const moduleFolderName = path.basename(moduleRootDir);
 console.log(`Linking ${targetDir} to ${moduleRootDir} as ${moduleFolderName}`);
 
+// Check if symlink already exists
+if (fs.existsSync(moduleFolderName)) {
+    console.log(`A file or directory with the name ${moduleFolderName} already exists in ${targetDir}.`);
+    process.exit(1);
+}
+
```

Comment:
Does this need to be an error? this is run as part of npm run devbuild and could just be the second time somebody ran devbuild.

Reply:
Correct, This should exit with 0